vendor/arm/mali/mali_profiler.cpp

```diff
@@ -47,13 +45,33 @@ struct MaliHWInfo
 	unsigned l2_slices;
 };
 
-MaliHWInfo get_mali_hw_info(const char *path)
+MaliHWInfo get_mali_hw_info(const MaliProfiler *profiler, const char *path)
 {
-	int fd = open(path, O_RDWR);        // NOLINT
+	struct ScopedFileHandle
+	{
+		int fd;
+		ScopedFileHandle(int fd) :
+		    fd(fd)
+		{}
+		~ScopedFileHandle()
+		{
+			if (fd >= 0)
+			{
+				close(fd);
+			}
+		}
+	};
+
+	MaliHWInfo hw_info;        // NOLINT
+	memset(&hw_info, 0, sizeof(hw_info));
+
+	int              fd = open(path, O_RDWR);        // NOLINT
```

Comment:
`// NOLINT` - what was the linter complaining about here?

Reply:
I'm unsure why there are `// NOLINT` comments throughout the project, This may of been a local setup of a previous developer which they committed to the repository. We could remove all `// NOLINT` comments?

Reply:
Removed all // NOLINT